miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/PoolPickerResponse.java

```diff
@@ -0,0 +1,91 @@
+package uk.ac.bbsrc.tgac.miso.webapp.util;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.ac.bbsrc.tgac.miso.core.util.PaginatedDataSource;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginationFilter;
+import uk.ac.bbsrc.tgac.miso.core.util.WhineyFunction;
+import uk.ac.bbsrc.tgac.miso.dto.PoolDto;
+import uk.ac.bbsrc.tgac.miso.dto.PoolOrderCompletionDto;
+
+public class PoolPickerResponse {
+
+  private static final Logger log = LoggerFactory.getLogger(PoolPickerResponse.class);
+
+  public static class PoolPickerEntry {
+    private final List<PoolOrderCompletionDto> orders;
+    private final PoolDto pool;
+
+    public PoolPickerEntry(PoolDto pool, List<PoolOrderCompletionDto> orders) {
+      super();
+      this.pool = pool;
+      this.orders = orders;
+    }
+
+    public List<PoolOrderCompletionDto> getOrders() {
+      return orders;
+    }
+
+    public PoolDto getPool() {
+      return pool;
+    }
+
+  }
+
+  List<PoolPickerEntry> items;
+  Long numMatches;
+  List<String> errors = new ArrayList<>();
+
+  public <T> void populate(PaginatedDataSource<T> source, boolean sortOrder, String sortColumn, Integer limit,
+      WhineyFunction<T, PoolPickerEntry> transform,
+      PaginationFilter... filters) throws IOException {
+
+    List<String> errors = new ArrayList<>();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "errors" which hides the field declared at line 45. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)